### PR TITLE
ModelClass get superClass() QOL method + W_M_Q.proto.class() rework CMDB-174

### PR DESCRIPTION
**Pull request by mingushead**
I certify that:
- [x] The automated build passes on the branch this pull request is from
- [x] My change is documented in the Changelist.md
- [ ] Any functionality additions/changes are documented in the README.md
- [ ] Any functionality additions/changes are _also_ documented [here](https://github.com/weaverplatform/weaver-docs/blob/master/pages/developers/reference/weaver-sdk-js.md "Weaver-docs")
- [x] Any new dockerimages this build depends on are properly tagged on github
- [x] I shall not merge until the travis build is green and the PR is accepted by
  another developer
- [x] I shall promptly fix/reply to comments, as I know PRs are not :fire: and
  forget
- [x] I shall keep my branch updated with any intermediate changes to the
  target branch
- [x] This is a quality PR
- [x] I got to the end of this list

### CODE REVIEW

**Comment by mingushead** [comment]
The use case is quite weak, it came up in a project I was working on, and seemed like something that would come up again.
My particular example was, _client needs a list of all the things that can have a document reference_.

In fact, I missed the fact that it only returned first-order stuff, I'm going to rework that now

**Comment by mingushead** [comment]
CMDB-174

File: src/WeaverModelQuery.coffee [hunk 1]
```diff
@@ -65,13 +65,16 @@ class WeaverModelQuery extends Weaver.Query
 
 
   class: (modelClass) ->
-    @hasRelationOut(@model.getMemberKey(), Weaver.Node.getFromGraph(modelClass.classId(), modelClass.context.getGraph()))
+    @preferredConstructor = modelClass
+    @hasRelationOut(@model.getMemberKey(),
+      new Weaver.ModelQuery(@model).hasRecursiveRelationOut(@model.getInheritKey(), Weaver.Node.getFromGraph(modelClass.classId(), modelClass.context.getGraph()), true)
+    )
```
Comment (gvdent) [inline src/WeaverModelQuery.coffee]: Sweet :)

File: src/WeaverModel.coffee [hunk 2]
```diff
@@ -246,6 +246,10 @@ class WeaverModel extends ModelContext
     console.warn('Deprecated function WeaverModel.getPrototypeKey() used. Use WeaverModel.getMemberKey().')
     @getMemberKey()
 
+  getIncludesKey: (str)->
+    for key,val of @includes
+      return key if val.definition.name is str
+
```
Comment (bastbijl) [inline src/WeaverModel.coffee]: What is this function for? Includes are registered as contexts.

Comment (bastbijl) [reply src/WeaverModel.coffee]: Actually, there is no way to get the key if you have the model name. So if you have usage for this function its a good one.

Comment (mingushead) [reply src/WeaverModel.coffee]: I didn't implement it this time round, but I was going to add in something like this to `WeaverModelClass`:
```coffee
  @getSubClasses: ->
    Object.keys(@model.classList).map((c)=>
      cPref = c.split(':')[0]
      cName = c.split(':')[1]
      prefix = @model.getIncludesKey(cPref)
      if prefix
        subC = @model[prefix][cName]
      else
        subC = @model[cName]```

Comment (mingushead) [reply src/WeaverModel.coffee]: Anyway, I'll remove `getIncludesKey` for now, as it isn't used in this pr

Comment (bastbijl) [reply src/WeaverModel.coffee]: That can be written different. Splitting on ':' is kind of forbidden. There are functions that should do that for you.

So maby this getIncludesKey too. So can you remove it for now?

Comment (mingushead) [reply src/WeaverModel.coffee]: removed

**Pull request merged**